### ISSUE TITLE
Pipeline deployment tested with MiniKube

### ISSUE THREAD

**Issue by gctucker** [kernelci/kernelci-api#273]
The first step with making a Kubernetes deployment is to look into the `kernelci-pipeline` services which come with a `docker-compose` setup.  These could be deployed in MiniKube while `kernelci-api` is running in a standard `docker-compose` deployment on the host.  That way, a generic Kubernetes deployment solution for the pipeline services can be created in a fully self-contained way before deploying it again in a real Cloud (e.g. Azure) as a follow-up task.

- [x] Local `kernelci-api` instance set up and running with standard `docker-compose` deployment
- [x] Running `kci api hello` on the host returns the "KernelCI API" message from the local instance
- [x] MiniKube set up locally on the same host
- [x] Pipeline services deployed in MiniKube so they can communicate with the local API instance
      Note: the `notifier` service is the easiest one to bring up so it can be done first to get started
      Note: the LAVA handler and KCIDB bridge defined in separate docker-compose files are out of scope for this issue
- [x] Configuration for automatically deploying all the pipeline services in Kubernetes tested with MiniKube
- [x] Configuration (Terraform?) added to the `kernelci-pipeline` repository

**Comment by JenySadadia** [comment]
Hello @sbdtu5498

Following the discussion in KernelCI community call and on https://github.com/kernelci/kernelci-api/issues/252, if you are available and willing to give a helping hand with k8s deployment, here is the initial issue to take a look at.
We can deploy the pipeline locally on Minikube as a first step.
If you are interested, I can also help to give some overview on API & Pipeline to get started with this.
Please let us know your thoughts and suggestions to take it further.

**Comment by sbdtu5498** [comment]
Sure thing, @JenySadadia! I will try to get it deployed this week if that's fine. I thought that testing with ACS started so this might be the final call. Regarding the API design, I took a brief overview and kind of got a hold of the stuff. I will create a local repo and will share the deployment files soon. Thanks!

**Comment by JenySadadia** [comment]
OK. Thanks @sbdtu5498

**Comment by gctucker** [comment]
@sbdtu5498 Great, thank you!  Also feel free to join the weekly meeting on Thursday again if you want to discuss this.  It's a public meeting, open to anyone who wants to contribute to KernelCI.

**Comment by sbdtu5498** [comment]
> @sbdtu5498 Great, thank you! Also, feel free to join the weekly meeting on Thursday again if you want to discuss this. It's a public meeting, open to anyone who wants to contribute to KernelCI.

Anytime!

**Comment by gctucker** [comment]
I've added a note to say that the LAVA callback handler and the KCIDB bridge services aren't part of this task as they're special cases and they're not part of the main pipeline.

Once this task is done, a possible follow-up would be this one to add the LAVA callback handler: https://github.com/kernelci/kernelci-api/issues/282

In a way it's the simplest service we need to deploy which needs to accept incoming HTTPS connections, so it might be good preparation work for then adding the API services (main API has HTTPS, then there's SSH and another HTTP for storage with static files).

**Comment by JenySadadia** [comment]
Hello @sbdtu5498 
How are things with this issue? Any pull request from you we can take a look at?

**Comment by sbdtu5498** [comment]
>  Pipeline services deployed in MiniKube so they can communicate with the local API instance

@gctucker this is possible but not the best idea due to security issues and increased complexity so I converted both the API deployments and Pipeline so that it can be natively deployed on the Kubernetes cluster. But this is still doable. Though, I do not see any benefits. Just read tit so thought of letting you know. Is there any specific thought that you have on this.

**Comment by gctucker** [comment]
I just thought it was a way to simplify the tasks by breaking it down into 2 steps.  Client services running in MiniKube should easily be able to connect to any API server so having them to run on the localhost in docker-compose should just work and not make any difference from the pipeline clients point of view.

**Comment by sbdtu5498** [comment]
Yeah, that might have been the case but just for testing purpose. Like we can simply do port-forward. But the abstractions in Kubernetes makes it a bit difficult to interact it locally with anything outside a Kubernetes Cluster from long term purpose. Anyways I will try to mention all the possible methods in README.

**Comment by JenySadadia** [comment]
Updated the checklist.

**Comment by JenySadadia** [comment]
Updated the checklist.
